scripts/version_check.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -e  # Exit on error
+
+# Ensure the script is executed from the repository root
+cd "$(dirname "$0")/.."
+
+PR_VERSION="${PR_VERSION}"
+DEV_VERSION="${DEV_VERSION}"
+MAIN_VERSION="${MAIN_VERSION}"
+BASE_BRANCH="${BASE_BRANCH}"
+
+echo "PR_VERSION=$PR_VERSION"
+echo "DEV_VERSION=$DEV_VERSION"
+echo "MAIN_VERSION=$MAIN_VERSION"
+echo "BASE_BRANCH=$BASE_BRANCH"
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "$SCRIPT_DIR/branch_type.sh"
+
+semver_gt() {
+  IFS='.' read -r major1 minor1 patch1 <<< "$1"
+  IFS='.' read -r major2 minor2 patch2 <<< "$2"
+
+  # Compare major version
+  if [ "$major1" -gt "$major2" ]; then
+    return 0
+  elif [ "$major1" -lt "$major2" ]; then
+    return 1
+  fi
+
+  # Compare minor version
+  if [ "$minor1" -gt "$minor2" ]; then
+    return 0
+  elif [ "$minor1" -lt "$minor2" ]; then
+    return 1
+  fi
+
+  # Compare patch version
+  if [ "$patch1" -gt "$patch2" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# Version Checking Logic
+if [ "$BASE_BRANCH" = "dev" ]; then
+  echo "PR into dev => applying dev rules"
+  if [ "$DEV_VERSION" = "$PR_VERSION" ]; then
+    echo "OK: dev version == PR version"
+    exit 0
+  else
+    if [ "$MAIN_VERSION" = "$DEV_VERSION" ]; then
+      if semver_gt "$PR_VERSION" "$DEV_VERSION"; then
+        echo "Bump dev"
```

Comment:
A more descriptive print text would be nice :) Maybe something like "OK: Increasing working version in dev from $DEV_VERSION to $PR_VERSION", or something like that